File: Napakalakirb/lib/monster.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "bad_consequence.rb"
require_relative "prize.rb"

class Monster
  attr_reader :name
  attr_reader :combatLevel
  attr_reader :malrollo
  
  def initialize(name , level , bc , p, ic = 0)
    @name = name
    @combatLevel = level
    @malrollo = bc
    @buenrollo = p
    @levelChangeAgainstCultistPlayer = ic
  end
  def getLevelsGained
    return @buenrollo.level
  end
  def getTreasuresGained
    return @buenrollo.treasures
  end
  def getLevelChangeAgainstCultistPlayer
    return @levelChangeAgainstCultistPlayer + @combatLevel
  end
  def to_s
    "Nombre: #{@name}, Nivel de combate: #{@combatLevel},
    Buen rollo: #{@buenrollo}
    Mal rollo: #{@malrollo}\n"
  end
end

File: Napakalaki/src/napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author jose
 */
public class Monster {
    private String name;
    private int level;
    private Prize buenrollo;
    private BadConsequence malrollo;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster( String name , int p_level , BadConsequence bc , Prize prize , int IC ){
        this.name = name;
        level = p_level;
        buenrollo = prize;
        malrollo = bc;
        levelChangeAgainstCultistPlayer = IC;
    }
    public Monster( String name , int p_level , BadConsequence bc , Prize prize){
        this.name = name;
        level = p_level;
        buenrollo = prize;
        malrollo = bc;
        levelChangeAgainstCultistPlayer = 0;
    }
    public int getCombatLevelAgainstCultistPlayer(){
        return level + levelChangeAgainstCultistPlayer;
    }
    public String getName(){
        return name;
    }
    public int getCombatLevel(){
        return level;
    }
    public BadConsequence getBadConsequence(){
        return malrollo;
    }
    public int getLevelsGained(){
        return buenrollo.getLevel();
    }
    public int getTreasuresGained(){
        return buenrollo.getTreasures();
    }
    public Prize getPrize(){
        return buenrollo;
    }
    public String toString(){
        return "Nombre: " + name + "\nNivel de combate: " + Integer.toString(level) + "\nNivel de combate SECTARIOS: " + getCombatLevelAgainstCultistPlayer() + "\n";
    }
}
